packages/firebase_storage/android/src/main/java/io/flutter/plugins/firebase/storage/FirebaseStoragePlugin.java

```diff
@@ -98,12 +138,18 @@ public void onFailure(Exception e) {
   }
 
   private void putFile(MethodCall call, final Result result) {
-    Map<String, String> arguments = (Map<String, String>) call.arguments;
-    String filename = arguments.get("filename");
-    String path = arguments.get("path");
+    Map<String, Object> arguments = (Map<String, Object>) call.arguments;
```

Comment:
Casting `call.arguments` leads to unnecessary fights with the type checker. Just do
```java
String filename = call.argument("filename");
String path = call.argument("path");
Map<String, Object> metadata = call.argument("metadata");
```